File: tourism-be/src/Modules/Blog/Explorer.Blog.Core/Domain/Comment.cs
using Explorer.BuildingBlocks.Core.Domain;
using System.Reflection.Metadata;

namespace Explorer.Blog.Core.Domain
{
    public class Comment : Entity
    {
        public long UserId { get; init; }
        public DateTime CreationDate { get; init; }
        public string Description { get; init; }
        public DateTime LastEditDate { get; init; }
        public long BlogId { get; init; }

        public Comment(long userId, DateTime creationDate, string description, DateTime lastEditDate, long blogId)
        {
            UserId = userId;
            CreationDate = creationDate;
            Description = description;
            LastEditDate = lastEditDate;
            BlogId = blogId;
            Validate();
        }

        private void Validate()
        {
            if (string.IsNullOrWhiteSpace(Description)) throw new ArgumentException("Invalid Description");
        }
    }
}

File: golang-be-blogs/model/Comment.go
package model

import "time"

type Comment struct {
	Id           int    `gorm:"primaryKey" json:"Id"`
	Description  string `json:"Description"`
	UserId       int    `json:"UserId"`
	BlogId       int    `json:"BlogId"`
	CreationDate time.Time `json:"CreationDate"`
	LastEditDate time.Time `json:"LastEditDate"`
}
